packages/web-components/fast-foundation/src/dialog/dialog.ts

```diff
@@ -85,6 +85,28 @@ export class Dialog extends FASTElement {
         this.$emit("dismiss");
     }
 
+    /**
+     * The method to show the dialog.
+     *
+     * @public
+     */
+    public show(): void {
+        if (this.hidden) {
```

Comment:
I actually think you can remove the guards here. It's safe to set it to the same value. The observation system will not trigger if there's no change. Not a big deal though.